webapp/sources/ldap-inventory/inventory-provisioning-web/src/main/scala/com/normation/inventory/provisioning/endpoint/InventoryProcessor.scala

```diff
@@ -114,123 +110,122 @@ class InventoryProcessor(
   , nodeInventoryDit: InventoryDit
 ) {
 
-  val logger = InventoryLogger
+  val logger = InventoryProcessingLogger
 
-  // current queue size
-  lazy val queueSize = AtomicInt(0)
+  /*
+   * We manage inventories buffering with a bounded queue.
+   * In case of overflow, we reject new inventories.
+   */
+  val queue = ZioRuntime.unsafeRun(Queue.dropping[InventoryReport](maxQueueSize))
```

Comment:
Inventory queue was managed by an AtomicInt+appropriate locking. We now quand just use an async queue with 'dropping' strategy. In place of checking if the AtomicInt is too big, we just check if `offer` returned true or false.